### PR TITLE
test: Add cosign and notations bats tests

### PR DESCRIPTION
**Pull request by nicoldr**
Signed-off-by: Nicol Draghici <idraghic@cisco.com>

**What type of PR is this?**

**Which issue does this PR fix**:
resolves: #867 

**What does this PR do / Why do we need it**:

**If an issue # is not available please add repro steps and logs showing the issue**:

**Testing done on this change**:

**Automation added to e2e**:

**Will this break upgrades or downgrades?**

**Does this PR introduce any user-facing change?**:

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -332,5 +332,5 @@ $(STACKER):
 	chmod +x $@
 
 .PHONY: test-annotations
-test-annotations: binary check-skopeo $(BATS) $(STACKER)
+test-annotations: binary check-skopeo $(BATS) $(STACKER) $(NOTATION)
```
Comment (andaaron) [inline Makefile:335]: How does cosign get installed? Don't we need to add that as a dependency as well?

Comment (nicoldr) [reply Makefile]: Yes, I realized that was the missing part. Just added it in the Makefile

**Review by andaaron — DISMISSED** [review]

**Review by rchincha — APPROVED** [review]
lgtm

**Comment by rchincha** [comment]
@nicoldr pls sign your commit and push

**Comment by nicoldr** [comment]
> 

Rebased and signed the commit

File: test/blackbox/annotations.bats [hunk 2]
```diff
@@ -89,3 +89,38 @@ function teardown_file() {
     [ $(echo "${lines[-1]}" | jq '.data.ImageList[0].Vendor') = '"CentOs"' ]
     [ $(echo "${lines[-1]}" | jq '.data.ImageList[0].Licenses') = '"GPLv2"' ]
 }
+
+@test "sign image with cosign" {
```
Comment (rchincha) [inline test/blackbox/annotations.bats:93]: "sign/verify with cosign"

File: test/blackbox/annotations.bats [hunk 3]
```diff
@@ -89,3 +89,38 @@ function teardown_file() {
...
+    run cosign sign --key cosign.key localhost:8080/annotations:latest --yes
+    [ "$status" -eq 0 ]
+    run cosign verify --key cosign.pub localhost:8080/annotations:latest
+    [ "$status" -eq 0 ]
+    local sigName=$(echo "${lines[-1]}" | jq '.[].critical.image."docker-manifest-digest"')
+    [ "$status" -eq 0 ]
+    [[ "$sigName" == *"${digest}"* ]]
+}
+
+@test "add signatures to image with notation" {
```
Comment (rchincha) [inline test/blackbox/annotations.bats:112]: "sign/verify with notation"

**Review by andaaron — APPROVED** [review]

**Review by rchincha — APPROVED** [review]
lgtm

**Pull request merged**